File: uber/uber_algo/LC125ValidPalindrome.py
class Solution(object):
    def isPalindrome(self, s):
        if not s:
            return True
        s = s.strip()
        left = 0
        right = len(s)-1
        while left <= right:
            if s[left].isalnum() and s[right].isalnum():
                if s[left].lower() != s[right].lower():
                    return False
                else:
                    left += 1
                    right -= 1
            elif not s[left].isalnum():
                left += 1
            elif not s[right].isalnum():
                right -= 1
        return True

File: apple/LC125ValidPalindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        s = s.toLowerCase();
        s = s.replaceAll("[^a-z^0-9]+","");
        int len = s.length();
        for (int i = 0; i < len; i++) {
            if (s.charAt(i) != s.charAt(len-i-1)) return false;
        }
        return true;
    }
}
